server/container/guice/cassandra/src/main/java/org/apache/james/modules/mailbox/CassandraCacheSessionModule.java

```diff
@@ -72,12 +71,8 @@ static class InitializedCacheCluster {
         private final CqlSession cluster;
 
         @Inject
-        private InitializedCacheCluster(CqlSession cluster, ClusterConfiguration clusterConfiguration, KeyspacesConfiguration keyspacesConfiguration) {
-            this.cluster = cluster;
-
-            if (clusterConfiguration.shouldCreateKeyspace()) {
-                KeyspaceFactory.createKeyspace(keyspacesConfiguration.cacheKeyspaceConfiguration(), cluster).block();
-            }
+        private InitializedCacheCluster(ResilientClusterProvider sessionProvider, KeyspacesConfiguration keyspacesConfiguration) {
+            this.cluster = sessionProvider.get(keyspacesConfiguration.cacheKeyspaceConfiguration());
```

Comment:
we don't create the keyspace anymore if it's necessary?

Reply:
Session provider creates it implicitly?